src/VirtoCommerce.ProfileExperienceApiModule.Data/Extensions/SchemaExtensions.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Threading.Tasks;
+using GraphQL;
+using GraphQL.Resolvers;
+using GraphQL.Types;
+using MediatR;
+using VirtoCommerce.ExperienceApiModule.Core.Extensions;
+using VirtoCommerce.ExperienceApiModule.Core.Helpers;
+using VirtoCommerce.ProfileExperienceApiModule.Data.Aggregates;
+using VirtoCommerce.ProfileExperienceApiModule.Data.Queries;
+using VirtoCommerce.ProfileExperienceApiModule.Data.Schemas;
+
+namespace VirtoCommerce.ProfileExperienceApiModule.Data.Extensions;
+
+public static class SchemaExtensions
+{
+    public static void AddMemberQuery<TAggregate, TType, TQuery>(this ISchema schema, IMediator mediator, string name, Func<string, object, Task> checkAuthAsync)
```

Comment:
Use Query base class from Xapi Core if you want to refactor schema
https://github.com/VirtoCommerce/vc-module-experience-api/blob/dev/src/VirtoCommerce.ExperienceApiModule.Core/BaseQueries/Query.cs
Example: https://github.com/VirtoCommerce/vc-module-quote/tree/dev/src/VirtoCommerce.QuoteModule.ExperienceApi/Queries

Reply:
I know and like it, but refactored queries has `CheckAuthAsync` call, which should be refactored first. It's calling in ProfileSchema **22 times**, so it's not something I want to refactor now. I avoided code duplication for newly introduced Vendor queries and introduced base queries for members, they can be inherited from base quires of XAPI later.